### ISSUE TITLE
scroll_page_count decimal value behaviour disagrees with documentation

### ISSUE THREAD

**Issue by indigophox** [irssi/irssi#1365]
https://irssi.org/documentation/settings/ indicates that for example using '.33' will result in a ~1/3-page scroll.  However .33 is resulting in 3 lines, .8 for example in 1 line, .44 in 2 lines...  Using '/3' works correctly however obviously the decimal value behaviour is not as-documented.

**Comment by ailin-nemui** [comment]
interesting find! 

the code is weird. if you give .xx, it calculates 1/.xx and uses that as the number of lines. I _think_ it might not be doing what was intended; maybe the settings docs are more true to the intention, in which case the code should be fixed?!

**Comment by ailin-nemui** [comment]
try this patch maybe?
```
diff --git a/src/fe-text/gui-readline.c b/src/fe-text/gui-readline.c
index 8e1b66ff..88ea254b 100644
--- a/src/fe-text/gui-readline.c
+++ b/src/fe-text/gui-readline.c
@@ -163,7 +163,7 @@ static int get_scroll_count(void)
 	else if (count < 1)
                 count = 1.0/count;
 
-	if (*str == '/') {
+	if (*str == '/' || *str == '.') {
 		count = (active_mainwin->height-active_mainwin->statusbar_lines)/count;
 	}
 	return (int)count;
```

**Comment by indigophox** [comment]
Agreed, regardless of which is more-right having it statically use 1/x is overall useless whereas what the documentation describes is useful behaviour (although obviously redundant vs. the fact that you can *also* use '/3' or whatever).  I suspect the documentation saying "about a third of a page" is probably because the 1 you bumped into is intended to be some window-height that might include title/input/status lines etc.

I'm unfortunately not very set-up to build locally for testing, this is MacOS via Brew and the Gentoo target I could test on needs some surgery before I can do so there :(

**Comment by indigophox** [comment]
Oh sorry, had my brain in database things yesterday.  Patch doesn't quite add up now that I re-read it.  Consider:

```
 	else if (count < 1)
-                count = 1.0/count;
+                count = (active_mainwin->height-active_mainwin->statusbar_lines)*count;
 
 	if (*str == '/') {
		count = (active_mainwin->height-active_mainwin->statusbar_lines)/count;
 	}
 	return (int)count;
```

Untested but this is probably what's needed.  It's that hardcoded 1.0 that makes the behaviour wrong AFAICT.
